### PR TITLE
Cleanup rs with timebound

### PR DESCRIPTION
**Pull request by anuddeeph1**
## Description
ReplicaSets serve as an intermediate controller for various Pod controllers like Deployments. When a new version of a Deployment is initiated, it generates a new ReplicaSet with the specified number of replicas and scales down the current one to zero. Consequently, numerous empty ReplicaSets may accumulate in the cluster, leading to clutter and potential false positives in policy reports if enabled. This cleanup policy is designed to remove empty ReplicaSets across the cluster within a specified timeframe, for instance, ReplicaSets created one day ago, ensuring the ability to rollback to previous ReplicaSets in case of deployment issues.

## Checklist
- [] I have read the [policy contribution guidelines](https://github.com/kyverno/policies/blob/main/README.md#contribution).
- [] I have added test manifests and resources covering both positive and negative tests that prove this policy works as intended.
- [] I have added the artifacthub-pkg.yml file and have verified it is complete and correct.

### CODE REVIEW

File: cleanup/cleanup-empty-replicasets/rbac.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+metadata:
+  name: kyvernocleanup-rs
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: kyverno:cleanup-rs
+subjects:
+- kind: ServiceAccount
+  name: kyverno-cleanup-controller
+  namespace: kyverno
```
Comment (chipzoller) [inline cleanup/cleanup-empty-replicasets/rbac.yaml:31]: This isn't necessary. As long as the appropriate labels are present, the permissions will be aggregated.

File: cleanup/cleanup-empty-replicasets/cleanup-empty-replicasets.yaml [hunk 2]
```diff
@@ -1,31 +1,35 @@
-apiVersion: kyverno.io/v2beta1
+apiVersion: kyverno.io/v2alpha1
```
Comment (chipzoller) [inline cleanup/cleanup-empty-replicasets/cleanup-empty-replicasets.yaml:1]: Why are you changing this?

File: cleanup/cleanup-empty-replicasets/cleanup-empty-replicasets.yaml [hunk 3]
```diff
@@ -1,31 +1,35 @@
...
 kind: ClusterCleanupPolicy
 metadata:
   name: cleanup-empty-replicasets
   annotations:
-    policies.kyverno.io/title: Cleanup Empty ReplicaSets
+    policies.kyverno.io/title: Cleanup Empty ReplicaSets
```
Comment (chipzoller) [inline cleanup/cleanup-empty-replicasets/cleanup-empty-replicasets.yaml:6]: Remove trailing space.

File: cleanup/cleanup-empty-replicasets/cleanup-empty-replicasets.yaml [hunk 4]
```diff
@@ -1,31 +1,35 @@
...
+        namespaces:
+          - kube-system
   conditions:
     all:
     - key: "{{ target.spec.replicas }}"
       operator: Equals
       value: 0
-  schedule: "*/5 * * * *"
+    - key: "{{ time_diff('{{target.metadata.creationTimestamp}}','{{ time_now_utc() }}') }}"
+      operator: "GreaterThan"
```
Comment (chipzoller) [inline cleanup/cleanup-empty-replicasets/cleanup-empty-replicasets.yaml:32]:
```suggestion
      operator: GreaterThan
```

File: cleanup/cleanup-empty-replicasets/cleanup-empty-replicasets.yaml [hunk 5]
```diff
@@ -1,31 +1,35 @@
...
+      value: "0h0m30s"
+  schedule: "*/1 * * * *"
+#The described logic currently deletes the ReplicaSets created 30 seconds ago. You can adjust this timeframe according to your specific requirements.
```
Comment (chipzoller) [inline cleanup/cleanup-empty-replicasets/cleanup-empty-replicasets.yaml:35]: Please place comments above the intended line, not below.

File: cleanup/cleanup-empty-replicasets/artifacthub-pkg.yml [hunk 6]
```diff
@@ -1,9 +1,9 @@
-name: cleanup-empty-replicasets
-version: 1.0.0
-displayName: Cleanup Empty ReplicaSets
-createdAt: "2023-12-12T17:57:28.000Z"
+name: cleanup-empty-replicasets-with-timebound
```
Comment (chipzoller) [inline cleanup/cleanup-empty-replicasets/artifacthub-pkg.yml:1]: Policy name has not changed. No need to change this.

File: cleanup/cleanup-empty-replicasets/artifacthub-pkg.yml [hunk 7]
```diff
@@ -1,9 +1,9 @@
...
+version: 1.0.1
+displayName: Cleanup empty replicasets with timebound
```
Comment (chipzoller) [inline cleanup/cleanup-empty-replicasets/artifacthub-pkg.yml:3]: Policy name has not changed. No need to change this.

**Review by chipzoller — CHANGES_REQUESTED** [review]

**Review by anuddeeph1 — COMMENTED** [review]
Updated the changes

**Comment by chipzoller** [comment]
When you make and complete requested changes, please resolve conversations as you go along so reviewers know you have addressed the comments.

**Review by chipzoller — CHANGES_REQUESTED** [review]
The `/cleanup` folder has not yet been activated for Chainsaw tests. Would you please also add the appropriate line [here](https://github.com/kyverno/policies/blob/main/.github/workflows/test.yml#L36-L37)?

**Review by chipzoller — APPROVED** [review]

**Pull request merged**